storage/streams/src/main/java/io/apicurio/registry/streams/utils/WaitForDataService.java

```diff
@@ -0,0 +1,140 @@
+package io.apicurio.registry.streams.utils;
+
+import io.apicurio.registry.storage.proto.Str;
+import io.apicurio.registry.streams.diservice.AsyncBiFunctionService;
+import io.apicurio.registry.utils.kafka.ProtoSerde;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.state.ReadOnlyKeyValueStore;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author Ales Justin
```

Comment:
A comment about what this does wouldn't hurt.

Reply:
Done.